frontend/src/features/workspace/components/WorkspaceHeader/WorkspaceHeader.tsx

```diff
@@ -36,139 +31,89 @@ export const WorkspaceHeader = ({
   handleOpenCreateFormModal,
 }: WorkspaceHeaderProps): JSX.Element => {
   const isMobile = useIsMobile()
+  const isDesktop = useMediaMatch(`(min-width: ${BREAKPOINT_VALS.lg})`)
+
   const {
     isLoading,
     totalFormsCount,
     displayedFormsCount,
+    activeSearch,
+    setActiveSearch,
     activeFilter,
     setActiveFilter,
   } = useWorkspaceContext()
-  const { isOpen, onClose, onOpen } = useDisclosure()
-
-  const mobileDrawerExtraButtonProps: Partial<ButtonProps> = useMemo(
-    () => ({
-      isFullWidth: true,
-      justifyContent: 'flex-start',
-      variant: 'clear',
-      colorScheme: 'secondary',
-      textStyle: 'body-1',
-    }),
-    [],
-  )
-
-  const handleSetActiveFilterFromDrawer = useCallback(
-    (filterOption: FilterOption | null) => {
-      setActiveFilter(filterOption)
-      onClose()
-    },
-    [onClose, setActiveFilter],
-  )
 
-  const renderFilterButton = useCallback(
-    (filterOption: FilterOption | null) => {
-      return (
-        <Stack
-          direction="row"
-          justify="space-between"
-          alignItems="center"
-          w="100%"
-        >
-          <Text>{filterOption ?? 'All forms'}</Text>
-          {activeFilter === filterOption ? <BiCheck /> : null}
-        </Stack>
-      )
-    },
-    [activeFilter],
-  )
+  const { isOpen: isSearchExpanded, onToggle: onToggleSearchExpansion } =
+    useDisclosure()
 
-  const headerStyle = useMemo(
-    () => (isMobile && activeFilter != null ? 'h3' : 'h2'),
-    [activeFilter, isMobile],
+  const headerText = useMemo(
+    () =>
+      activeSearch || activeFilter
+        ? simplur`Showing ${displayedFormsCount} of ${totalFormsCount} form[|s]`
+        : `All forms (${totalFormsCount})`,
+    [activeFilter, activeSearch, displayedFormsCount, totalFormsCount],
   )
 
   return (
     <Grid
       gridTemplateAreas={{
-        base: "'header filter' 'create create'",
-        md: "'header filter create'",
+        base: isSearchExpanded
+          ? "'header searchicon filter' 'search search search' 'create create create'"
+          : "'header searchicon filter' 'create create create'",
+        md: isSearchExpanded
+          ? "'header searchicon filter create' 'search search search search'"
+          : "'header searchicon filter create'",
+        lg: "'header searchfilter create'",
```

Comment:
nit: use camelcase for grid areas for readability? or is all lowercase a convention

Reply:
will update, though it seems that everywhere else we use lowercase haha. can update elsewhere whenever else